velox/dwio/common/ScanSpec.cpp

```diff
@@ -142,7 +142,7 @@ bool ScanSpec::hasFilter() const {
   if (hasFilter_.has_value()) {
     return hasFilter_.value();
   }
-  if (!isConstant() && filter()) {
+  if (filter()) {
```

Comment:
`hasFilter_` should be true for constant as `testFilterOnConstant` evaluates the filter on constant. This enables filter push-down for the constant NULL generated by schema evolution.

https://github.com/facebookincubator/velox/blob/192531c7c3b2ad0044d1a56c2a52c3b49fdfdfa9/velox/dwio/common/SelectiveStructColumnReader.cpp#L26-L34

Reply:
@rui-mo Can we move this to a separate PR since this fix is independent? Thanks.

Reply:
Never mind. I see that the schema evolution test added here introduces this NULL.